### PR TITLE
CI: add github action to build source release tarball

### LINKED ISSUE #1315: wabt fails to build from source because of missing wasm.h
When I try to build wabt 1.0.13 from source, the build fails because wasm.h is not present in the source package.

I know that wasm.h is present in binaryen, but I don't understand how 

src/interp/interp-wasm-c-api.cc

can be built from source in wabt without wasm.h.

In version 1.0.12 wasm.h was not required to compile interp-wasm-c-api.cc.

Please clarify how to build wabt from source in version 1.0.13

### LINKED ISSUE #1367: build failed: fatal error: wasm.h : No such file or directory
## OS Info
```bash
➜  wabt git:(master) cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

➜  wabt git:(master) gcc --version
gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

➜  wabt git:(master) clang --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

## build steps

follow the build steps in the [Building using the top-level Makefile (Linux and macOS)](https://github.com/webassembly/wabt#building-using-the-top-level-makefile-linux-and-macos)

make command output:

```bash
➜  wabt git:(master) make gcc-release-no-tests
mkdir -p out/gcc/Release/no-tests/
cd out/gcc/Release/no-tests/ && cmake -G "Unix Makefiles" /home/flyflypeng/Code/Github/wabt/ -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=OFF
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Check size of size_t
-- Check size of size_t - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/flyflypeng/Code/Github/wabt/out/gcc/Release/no-tests
make --no-print-directory -C out/gcc/Release/no-tests/ all
...... // above everything is ok
[ 84%] Building CXX object CMakeFiles/wasm.dir/src/wast-parser.cc.o
/home/flyflypeng/Code/Github/wabt/src/wast-parser.cc: In member function ‘wabt::Result wabt::WastParser::ParseConst(wabt::Const*, wabt::WastParser::ConstType)’:
/home/flyflypeng/Code/Github/wabt/src/wast-parser.cc:2262:14: warning: ‘literal’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       result = ParseDouble(literal.type, s, end, &f64_bits);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/flyflypeng/Code/Github/wabt/src/wast-parser.cc:2262:14: warning: ‘end’ may be used uninitialized in this function [-Wmaybe-uninitialized]
/home/flyflypeng/Code/Github/wabt/src/wast-parser.cc:2262:14: warning: ‘s’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[ 85%] Building CXX object CMakeFiles/wasm.dir/src/wat-writer.cc.o
[ 86%] Building CXX object CMakeFiles/wasm.dir/src/interp/binary-reader-interp.cc.o
[ 87%] Building CXX object CMakeFiles/wasm.dir/src/interp/interp.cc.o
[ 87%] Building CXX object CMakeFiles/wasm.dir/src/interp/interp-util.cc.o
[ 88%] Building CXX object CMakeFiles/wasm.dir/src/interp/istream.cc.o
[ 89%] Building CXX object CMakeFiles/wasm.dir/src/interp/interp-wasm-c-api.cc.o
/home/flyflypeng/Code/Github/wabt/src/interp/interp-wasm-c-api.cc:17:10: fatal error: wasm.h: No such file or directory
 #include <wasm.h>
          ^~~~~~~~
compilation terminated.
CMakeFiles/wasm.dir/build.make:1094: recipe for target 'CMakeFiles/wasm.dir/src/interp/interp-wasm-c-api.cc.o' failed
make[3]: *** [CMakeFiles/wasm.dir/src/interp/interp-wasm-c-api.cc.o] Error 1
CMakeFiles/Makefile2:591: recipe for target 'CMakeFiles/wasm.dir/all' failed
make[2]: *** [CMakeFiles/wasm.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make[1]: *** [all] Error 2
Makefile:170: recipe for target 'gcc-release-no-tests' failed
make: *** [gcc-release-no-tests] Error 2
```

So I think should add wasm.h header file in the CMakeList.txt file.

### LINKED ISSUE #1594: src/interp/interp-wasm-c-api.cc:17:10: fatal error: 'wasm.h' file not found
I have a problem in MAC, operation
cmake -- build .
thanks!

### PR DESCRIPTION
**Pull request by rofl0r**
closes #1315

result can be seen here
https://github.com/rofl0r/wabt/releases/tag/9.9.99d

### CODE REVIEW

File: .github/workflows/build_source_release.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+# Trigger whenever a release is created
+on:
+  release:
+    types:
+      - created
+
+jobs:
+  build:
+    name: build
```
Comment (sbc100) [inline .github/workflows/build_source_release.yml:11]: Could we just add this as a second job called "build-source-release" to the existing `build_release.yml`?

The jobs will run in parallel, and will ensure that they both get triggered on the same events.

File: .github/workflows/build_source_release.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
```
Comment (sbc100) [inline .github/workflows/build_source_release.yml:15]: Can we just do `runs-on: ubuntu-latest` and skip the strategy matrix?

File: .github/workflows/build_source_release.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    defaults:
+      run:
+        shell: bash
```
Comment (sbc100) [inline .github/workflows/build_source_release.yml:18]: I don't think this block is needed since bash is the default on linux.

**Review by sbc100 — COMMENTED** [review]
Nice!  Thanks for working on this.  

lgtm % nits/comments

**Comment by sbc100** [comment]
It seems that neither #1367 nor #1594 are actually addressed by this fix since they both involve folks who are using `git clone` themselves.    This fix only effects users who download source archives, not those who use git, right?

File: .github/workflows/build_source_release.yml (hunk 1 above)
Comment (rofl0r) [reply .github/workflows/build_source_release.yml:11]: i was looking into that earlier, but i'm not familiar with github workflows and shoehorning it into the existing script seemed to involve either arcane knowledge or adding a lot of if: special cases to the existing targets. having it as a separate task seems to be much cleaner, and allowed me to come up with a solution to begin with. both tasks do actually run in parallel in my tests, and the build-source-release tasked finished much faster (naturally, as it doesn't need to compile C++ code...)

File: .github/workflows/build_source_release.yml (hunk 2 above)
Comment (rofl0r) [reply .github/workflows/build_source_release.yml:15]: maybe, i would need to try it out.

File: .github/workflows/build_source_release.yml (hunk 3 above)
Comment (rofl0r) [reply .github/workflows/build_source_release.yml:18]: ok, will amend after a test.

**Comment by rofl0r** [comment]
> It seems that neither #1367 nor #1594 are actually addressed by this fix since they both involve folks who are using git clone themselves. This fix only effects users who download source archives, not those who use git, right?

correct. i didnt look into those issues tbh, and assumed they would have hit the same issue. will edit introductory text accordingly.

**Comment by rofl0r** [comment]
seems to have worked out: https://github.com/rofl0r/wabt/releases/tag/9.9.99e

i can squash the follow-up commit, or you can do so using the "rebase-and-merge" button.

File: .github/workflows/build_source_release.yml (hunk 1 above)
Comment (sbc100) [reply .github/workflows/build_source_release.yml:11]: OK we can leave it separate for now

**Comment by sbc100** [comment]
We always use squash commits anyway so it good as is.

**Comment by sbc100** [comment]
Once this lands we can tag a release to show this working..

**Review by sbc100 — APPROVED** [review]

**Pull request merged**